### PR TITLE
[RNBW-2567] Add/Remove Favorites Inconsistencies on CurrencySelectModal

### PR DESCRIPTION
**Pull request by derHowie**
Fixes RNBW-2567

## What changed (plus any additional context for devs)
Selecting and deselecting favorites became inconsistent. Sometimes upon navigating back to the currency select modal the user's selections would be reflected in the favorites list and others times that would not be the case. This issue was made worse if a user attempted to remove/add many assets at once.

## PoW (screenshots / screen recordings)
https://user-images.githubusercontent.com/14877580/155601887-85e3856d-d073-4e6a-8936-0676baecb4d6.mp4

## Dev checklist for QA: what to test
Ensure that adding/removing favorites on the currency select modal functions correctly.

## Final checklist

- [x] Assigned individual reviewers?
- [x] Added labels?

### CODE REVIEW

**Review by skylarbarrera — APPROVED** [review]
👑

File: src/screens/CurrencySelectModal.js [hunk 1]
```diff
@@ -186,13 +186,22 @@ export default function CurrencySelectModal() {
...
+        } else {
+          remove.push(current);
+        }
+        return [add, remove];
+      },
+      [[], []]
+    );
+    await updateFavorites(assetsToAdd, true);
+    await updateFavorites(assetsToRemove, false);
+  }, [assetsToFavoriteQueue, updateFavorites]);
```
Comment (brunobar79) [inline src/screens/CurrencySelectModal.js:204]: I'd move this into a then syntax so it's not blocking. Something around the lines of

`updateFavorites(assetsToAdd, true).then ( () => updateFavorites(assetsToRemove, false));`

**Review by brunobar79 — APPROVED** [review]
👍 👍 👍

File: src/screens/CurrencySelectModal.js (hunk 1 above)
Comment (derHowie) [reply src/screens/CurrencySelectModal.js:205]: Alright, done. Did some testing lg2m.

**Review by ibrahimtaveras00 — APPROVED** [review]
Fix looks good on both OS's, QA Passed 👍🏽

**Pull request merged**